agent/engine/service_connect/manager_linux_test.go

```diff
@@ -189,3 +190,66 @@ func TestPauseContainerModificationsForServiceConnect(t *testing.T) {
 		})
 	}
 }
+
+func TestAgentContainerModificationsForServiceConnect(t *testing.T) {
+	scTask, _, serviceConnectContainer := getAWSVPCTask(t)
+
+	tempDir := t.TempDir()
+	expectedBinds := []string{
+		fmt.Sprintf("%s/status/%s:%s", tempDir, scTask.GetID(), "/some/other/run"),
+		fmt.Sprintf("%s:%s", tempDir, "/not/var/run"),
+	}
+	expectedENVs := map[string]string{
+		"ReLaYgOeShErE":  "/not/var/run/relay_file_of_holiness",
+		"StAtUsGoEsHeRe": "/some/other/run/status_file_of_holiness",
+	}
+
+	type testCase struct {
+		name          string
+		container     *apicontainer.Container
+		expectedENV   map[string]string
+		expectedBinds []string
+	}
+	testcases := []testCase{
+		{
+			name:          "Service connect container has extra binds/ENV",
+			container:     serviceConnectContainer,
+			expectedENV:   expectedENVs,
+			expectedBinds: expectedBinds,
+		},
+	}
+	for _, container := range scTask.Containers {
+		addDefaultCase := true
+		for _, tc := range testcases {
+			if tc.container == container {
+				addDefaultCase = false
+				break
+			}
+		}
+		if addDefaultCase {
+			testcases = append(testcases, testCase{name: container.Name, container: container, expectedENV: map[string]string{}})
+		}
```

Comment:
If no one other than the SC container should get the bind mounts and env vars, can we do `if a container is not SC container, add a default test case for it`?